array_analyzer/extract/txt_parser.py

```diff
@@ -38,34 +53,124 @@ def create_xml_dict(path_):
+
+        # fiducials
+        fiduc = layout['marker']
+
+        # spot IDs
+        spots = doc['configuration']['well_configurations']['configuration']['array']['spots']['spot']
 
-    # fiducials
-    fiduc = layout['marker']
+        # replicates
+        repl = doc['configuration']['well_configurations']['configuration']['array']['spots']['multiplet']
 
-    # spot IDs
-    spots = doc['configuration']['well_configurations']['configuration']['array']['spots']['spot']
+        array_params = dict()
+        array_params['rows'] = int(layout['@rows'])
+        array_params['columns'] = int(layout['@cols'])
+        array_params['v_pitch'] = float(layout['@vspace'])
+        array_params['h_pitch'] = float(layout['@hspace'])
+        array_params['spot_width'] = float(layout['@expected_diameter'])
+        array_params['bg_offset'] = float(layout['@background_offset'])
+        array_params['bg_thickness'] = float(layout['@background_thickness'])
+        array_params['max_diam'] = float(layout['@max_diameter'])
+        array_params['min_diam'] = float(layout['@min_diameter'])
 
-    # replicates
-    repl = doc['configuration']['well_configurations']['configuration']['array']['spots']['multiplet']
+        # not present in .xml but manually added here:
+        array_params['pixel_size_scienion'] = 0.0049
+        array_params['pixel_size_octopi'] = 0.00185
+    except Exception as ex:
+        raise AttributeError(f"exception while parsing .xml : {ex}")
 
-    params = dict()
-    params['rows'] = int(layout['@rows'])
-    params['columns'] = int(layout['@cols'])
-    params['v_pitch'] = float(layout['@vspace'])
-    params['h_pitch'] = float(layout['@hspace'])
-    params['spot_width'] = float(layout['@expected_diameter'])
-    params['bg_offset'] = float(layout['@background_offset'])
-    params['bg_thickness'] = float(layout['@background_thickness'])
-    params['max_diam'] = float(layout['@max_diameter'])
-    params['min_diam'] = float(layout['@min_diameter'])
+    return fiduc, spots, repl, array_params
 
-    # todo: add param for different imaging conditions (magnificaiton, camera pixel size, etc..)
-    params['pixel_size_scienion'] = 0.0049  # scienion camera
-    params['pixel_size_octopi'] = 0.00185  # octopi camera
 
-    return fiduc, spots, repl, params
+def create_csv_dict(path_):
+    """
+    Looks for three .csv files:
+        "array_parameters.csv" contains array printing parameters as well as hardware parameters
+        "array_format_type.csv" contains fiducial and control names, locations
+        "array_format_antigen.csv" contains specific antigen names for only diagnostic spots
+    Then, parses the .csvs and creates the four dictionaries:
+    :param path_: list
+        of strings to the .csv paths
+    :return:
+    """
+    # assign names to each of the types == params, spot types, antigens
+    fiduc = list()
+    csv_antigens = list()
+    array_params = dict()
+
+    for meta_csv in path_:
```

Comment:
I believe you've previously checked that your dir contains the three separate csv files so you could just specifically read each of them instead of using elif statements...